src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -353,7 +354,24 @@ def validate_display_type(self, display_type):
         return DashboardWidgetDisplayTypes.get_id_for_type_name(display_type)
 
     def validate_widget_type(self, widget_type):
-        return DashboardWidgetTypes.get_id_for_type_name(widget_type)
+        widget_type = DashboardWidgetTypes.get_id_for_type_name(widget_type)
+        if widget_type == DashboardWidgetTypes.DISCOVER or widget_type is None:
+            sentry_sdk.set_context(
+                "dashboard",
+                {
+                    "org_slug": self.context["organization"].slug,
+                },
+            )
+            sentry_sdk.capture_message("Created or updated widget with discover dataset.")
+            if features.has(
+                "organizations:deprecate-discover-widget-type",
+                self.context["organization"],
+                actor=self.context["request"].user,
```

Comment:
Oh, a bunch of tests are failing because the request isn't being passed down in context for the widget validation endpoint at `src/sentry/api/endpoints/organization_dashboard_widget_details.py`. If you just add `"request": request` on line 39 or so then this should work, or pass down the user directly